pkg/kn/commands/service/service_update.go

```diff
@@ -50,6 +50,10 @@ func NewServiceUpdateCommand(p *commands.KnParams) *cobra.Command {
 				return err
 			}
 
+			if cmd.Flags().NFlag() == 0 {
```

Comment:
The other option would be to do this in a `PreRun` function which should allow separation of the checks to the `Run`. See this [comment](https://github.com/spf13/cobra/issues/206) from Cobra community.

Since this check is simple, it's likely OK here but if we start adding more such checks or other commands then `PreRun` could be a nice way to write these pre-condition code. Cheers :)

Reply:
@maximilien As you mentioned, there may be more checks needed later, and people even mentioned about the odd error message when wrong args here # 317, so I moved this part to `PreRun`. Thanks!